platform/darwin/src/NSDictionary+MGLAdditions.mm

```diff
@@ -0,0 +1,42 @@
+#import "NSDictionary+MGLAdditions.h"
+
+#import "NSExpression+MGLAdditions.mm"
+
+@implementation NSDictionary (MGLAdditions)
+
+- (mbgl::PropertyMap)mgl_propertyMap {
+    mbgl::PropertyMap propertyMap;
+    for (NSString *key in self.allKeys) {
+        if ([self[key] isKindOfClass:[NSDictionary class]]) {
+            propertyMap[[key UTF8String]] = [self[key] mgl_propertyMap];
+        } else if ([self[key] isKindOfClass:[NSArray class]]) {
+            NSArray *array = self[key];
+            propertyMap[[key UTF8String]] = [self vectorForArray:array];
+        } else {
+            NSExpression *expression = [NSExpression expressionForConstantValue:self[key]];
+            propertyMap[[key UTF8String]] = [expression mgl_filterValue];
+        }
+    }
+    return propertyMap;
+}
+
+- (std::vector<mbgl::Value>)vectorForArray:(NSArray *)array {
+    std::vector<mbgl::Value> vector;
+    vector.reserve(array.count);
+    for (id value in array) {
+        if ([value isKindOfClass:[NSArray class]]) {
```

Comment:
Once we refactor this so that any object knows how to "valueize" itself, we'll be able to remove these conditionals too.